### PR TITLE
Feat/implementation of opensearch

### PR DESCRIPTION
**Pull request by gecBurton**
## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [ ] I have run integration tests https://github.com/i-dot-ai/redbox/actions/runs/12016708858

### CODE REVIEW

File: redbox-core/redbox/loader/ingester.py [hunk 1]
```diff
@@ -22,31 +20,13 @@
 alias = env.elastic_chunk_alias
 
 
-def get_elasticsearch_store(es, es_index_name: str):
```
Comment (gecBurton) [inline redbox-core/redbox/loader/ingester.py:25]: moved to settings.py

File: redbox-core/redbox/loader/ingester.py [hunk 2]
```diff
@@ -22,31 +20,13 @@
...
-    return ElasticsearchStore(
-        index_name=es_index_name,
-        embedding=get_embeddings(env),
-        es_connection=es,
-        query_field="text",
-        vector_query_field=env.embedding_document_field_name,
-    )
-
-
-def get_elasticsearch_store_without_embeddings(es, es_index_name: str):
```
Comment (gecBurton) [inline redbox-core/redbox/loader/ingester.py:35]: moved to settings.py

File: redbox-core/redbox/models/settings.py [hunk 3]
```diff
@@ -19,7 +24,11 @@ class OpenSearchSettings(BaseModel):
 
     model_config = SettingsConfigDict(frozen=True)
 
-    collection_endpoint: str
+    host: str
```
Comment (gecBurton) [inline redbox-core/redbox/models/settings.py:27]: @nboyse notice this change, is this ok?

Comment (nboyse) [reply redbox-core/redbox/models/settings.py:27]: Would host be most appropriate or maybe just endpoint? As it also contains creds for DSIT and DBT, but I guess not streamlined with other configs

Comment (nboyse) [reply redbox-core/redbox/models/settings.py:27]: I was thinking something along the lines of this, but maybe too complex for use case:
class OpenSearchSettings(BaseModel):
    """settings required for a aws/opensearch"""

    model_config = SettingsConfigDict(frozen=True)

    collection_endpoint: str = env.str("COLLECTION_ENDPOINT")
    parsed_url: AnyUrl = urlparse(collection_endpoint)

    collection_endpoint__username: Optional[str] = parsed_url.username
    collection_endpoint__password: Optional[str] = parsed_url.password
    collection_endpoint__host: Optional[str] = parsed_url.hostname
    collection_endpoint__port: Optional[str] = parsed_url.port

Comment (gecBurton) [reply redbox-core/redbox/models/settings.py:27]: oh i see, in that case how about?

```python
class OpenSearchSettings(BaseModel):
    """settings required for a aws/opensearch"""

    model_config = SettingsConfigDict(frozen=True)
    connection_string: str

    @computed_field
    @property
    def username(self) -> int:
        return urlparse(self.collection_endpoint). username

    @computed_field
    @property
    def password(self) -> int:
        return urlparse(self.collection_endpoint). password

    @computed_field
    @property
    def password(self) -> int:
        return urlparse(self.collection_endpoint). password

    @computed_field
    @property
    def port(self) -> int:
        return urlparse(self.collection_endpoint). port

```

**Pull request closed without merging**